ci_framework/roles/reproducer/tasks/main.yml

```diff
@@ -15,42 +15,114 @@
+  ansible.builtin.file:
+    path: "{{ cifmw_reproducer_basedir }}"
+    state: directory
+  loop:
+    - artifacts
+    - logs
+
+- name: Load CI job environment
+  when:
+    - cifmw_job_uri is defined
+  ansible.builtin.import_tasks: ci_data.yml
+
 - name: Deploy layout on target host
+  tags:
+    - bootstrap
   ansible.builtin.import_role:
     name: libvirt_manager
     tasks_from: deploy_layout
 
-- name: Ensure crc-0 knows about its second NIC
+- name: Push generated inventory from hypervisor
+  tags:
+    - bootstrap
+  ansible.builtin.command:  # noqa: command-instead-of-module
+    cmd: >-
+      rsync -r {{ cifmw_reproducer_basedir }}/reproducer-inventory/
+      zuul@controller-0:reproducer-inventory
+
+- name: Slurp ssh key for CRC access
+  tags:
+    - bootstrap
+  register: crc_priv_key
+  ansible.builtin.slurp:
+    path: ".crc/machines/crc/id_ecdsa"
+
+- name: Configure CRC services
+  tags:
+    - bootstrap
   delegate_to: crc-0
   remote_user: root
-  community.general.nmcli:
-    autoconnect: true
-    conn_name: private_net
-    ifname: enp2s0
-    type: ethernet
-    ip4: "{{ cifmw_reproducer_crc_ip4 }}"
-    gw4: "{{ cifmw_reproducer_crc_gw4 }}"
-    state: present
+  block:
+    - name: Ensure crc-0 knows about its second NIC
+      when:
+        - cifmw_job_uri is undefined
+      community.general.nmcli:
+        autoconnect: true
+        conn_name: private_net
+        dns4: 127.0.0.1
+        ifname: enp2s0
+        type: ethernet
+        ip4: "{{ cifmw_reproducer_crc_ip4 }}/24"
+        gw4: "{{ cifmw_reproducer_crc_gw4 }}"
+        state: present
+
+    - name: Ensure crc-0 does not get "public" DNS
+      community.general.nmcli:
+        autoconnect: true
+        conn_name: "Wired connection 1"
+        dns4_ignore_auto: true
+        state: present
+
+    - name: Configure dns forwarders
+      ansible.builtin.blockinfile:
+        path: "/var/srv/dnsmasq.conf"
+        block: |-
+          server=1.1.1.1
+          server=8.8.8.8
+
+    - name: Configure local DNS for CRC pod
+      register: last_modification
+      ansible.builtin.replace:
+        path: "/var/srv/dnsmasq.conf"
+        regexp: "192.168.130.11"
+        replace: "{{ cifmw_reproducer_crc_ip4 }}"
+
+    - name: Reboot CRC node
+      when:
+        - last_modification is changed
+        - cifmw_job_uri is undefined
+      ansible.builtin.reboot:
 
 - name: Get kubeconfig file from crc directory
+  tags:
+    - bootstrap
   register: kubeconfig
   ansible.builtin.slurp:
     path: "{{ cifmw_reproducer_kubecfg }}"
 
 - name: Configure controller-0
+  tags:
+    - bootstrap
   delegate_to: controller-0
   remote_user: root
   block:
     - name: Ensure controller-0 has an IP in private network
+      when:
+        - cifmw_job_uri is undefined
       community.general.nmcli:
         autoconnect: true
         conn_name: private_net
-        ifname: eth1
+        ifname: "{{ cifmw_reproducer_private_nic }}"
```

Comment:
How difficult is to avoid hard-coding this? I mean, maybe it's doable if we fetch the mac from libvirt and use that value to match the interface by calling `ip link`

Reply:
to be fair, I didn't want to get that whole `ip link -o json` thing in the game. In this precise case, it's almost hard-coded at the VM level - it may change once we get to PSI, but that's about it. libvirt, at least with the domain definition and options set in the libvirt_manager, are mostly fixed with that kind of names with the default centos image.